website/common/script/ops/scoreTask.js

```diff
@@ -243,11 +244,30 @@ module.exports = function scoreTask (options = {}, req = {}) {
           if (user.addNotification) user.addNotification('STREAK_ACHIEVEMENT');
         }
         task.completed = true;
+
+        // Save history entry for daily
+        task.history = task.history || [];
+        let historyEntry = {
+          date: Number(new Date()),
+          value: task.value,
+        };
+        task.history.push(historyEntry);
       } else if (direction === 'down') {
         // Remove a streak achievement if streak was a multiple of 21 and the daily was undone
         if (task.streak % 21 === 0) user.achievements.streak = user.achievements.streak ? user.achievements.streak - 1 : 0;
         task.streak -= 1;
         task.completed = false;
+
+        // Delete history entry when daily unchecked
+        if (task.history || task.history.length > 0) {
+          task.history.forEach((entry, index) => {
```

Comment:
Is it necessary to iterate over all the history entries here? Can't you check just the last entry since dailies have only one entry per day?

Reply:
Ok sure, wasn't sure about that but yes, I believe it's better to just get the last one then.
I'll change that 😊